test/guest-test.js

```diff
@@ -81,15 +81,15 @@ describe('Guest', () => {
   });
 
   it('should apologize profusely and give suggestions to proceed if there are no available rooms', () => {
-    expect(guest.filterRoomsByTypeOnDate('single room', "2020/06/27")).to.equal(
+    expect(guest.filterRoomsByTypeOnDate('2020/06/27', 'single room')).to.equal(
```

Comment:
I am not going to worry about the tests yet, as I know I have to do a bit of refactoring for several of them.